src/crypto/qat/qcccrypto.cc

```diff
@@ -66,7 +66,7 @@ auto QccCrypto::async_get_instance(CompletionToken&& token) {
       // keep a few objects to wait QAT instance to make sure qat full utilization as much as possible,
       // that is, QAT don't need to wait for new objects to ensure
       // that QAT will not be in a free state as much as possible
-      instance_completions.push_back([this, ex, handler2 = std::move(handler1)](int inst)mutable{
```

Comment:
does removing "this" from the lambda capture have any discrete purpose in the PR?

Reply:
the first 3 commits resolve compile warnings/errors. i should have documented those in the commit messages. this one only happens on clang:
```
[17/19] Building CXX object src/crypto/qat/CMakeFiles/ceph_crypto_qat.dir/qcccrypto.cc.o
src/crypto/qat/qcccrypto.cc:68:39: warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
   68 |       instance_completions.push_back([this, ex, handler2 = std::move(handler1)](int inst)mutable{
      |                                       ^
```